qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,134 @@
+import re
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest.utils_version import VersionInterval
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    cpuinfo query test:
+    1). run query cmd. e.g -cpu ?cpuid
+    2). check the expected info is included in the cmd output.
+    3). Boot guest and check the output of qmp command "qom-list-types"
+    4). Check the output of qmp command "query-cpu-definitions"
+    5). Check the output of qmp command "query-cpu-model-expansion"
+    """
+
+    cpu_types = ["Opteron_G5", "Opteron_G4", "Opteron_G3",
+                 "Opteron_G2", "Opteron_G1", "Skylake-Server",
+                 "Skylake-Client", "Broadwell", "Broadwell-noTSX",
+                 "Haswell", "Haswell-noTSX", "IvyBridge",
+                 "SandyBridge", "Westmere", "Nehalem",
+                 "Penryn", "Conroe"]
+    cpu_type_re = {"EPYC-Rome": "rdpid,wbnoinvd,stibp,clwb,umip",
+                   "EPYC": "avx2,adx,bmi2,sha_ni",
+                   "Opteron_G5": "f16c,fma4,xop,tbm",
+                   "Opteron_G4": ("fma4,xop,avx,xsave,aes,sse4.2|sse4_2,"
+                                  "sse4.1|sse4_1,cx16,ssse3,sse4a"),
+                   "Opteron_G3": "cx16,sse4a",
+                   "Opteron_G2": "cx16",
+                   "Opteron_G1": "",
+                   "KnightsMill": "avx512_4vnniw,avx512pf,avx512er",
+                   "Icelake-Server": "avx512_vnni,la57,clflushopt",
+                   "Icelake-Client": ("avx512_vpopcntdq|avx512-vpopcntdq,"
+                                      "avx512vbmi,avx512_vbmi2|avx512vbmi2,"
+                                      "gfni,vaes,vpclmulqdq,avx512_vnni"),
+                   "Cascadelake-Server": ("avx512f,avx512dq,avx512bw,avx512cd,"
+                                          "avx512vl,clflushopt,avx512_vnni"),
+                   "Skylake-Server": "avx512f,clwb,xgetbv1,pcid",
+                   "Skylake-Client": "xgetbv1,pcid",
+                   "Broadwell": "adx,rdseed,3dnowprefetch,hle",
+                   "Broadwell-noTSX": "adx,rdseed,3dnowprefetch",
+                   "Haswell": "fma,avx2,movbe,hle",
+                   "Haswell-noTSX": "fma,avx2,movbe",
+                   "IvyBridge": "f16c,fsgsbase,erms",
+                   "SandyBridge": ("avx,xsave,aes,sse4_2|sse4.2,sse4.1|sse4_1,"
+                                   "cx16,ssse3"),
+                   "Westmere": "aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                   "Nehalem": "sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                   "Penryn": "sse4.1|sse4_1,cx16,ssse3",
+                   "Conroe": "ssse3"}
```

Comment:
Hi, I update the cpu models because of some change of flags(https://github.com/avocado-framework/avocado-vt/pull/2672), please update models in here, thanks.

Reply:
@nanliu-r Yes, I could update it when the above pull request get merged, please help check other code before that, thanks.